File: conversation-service/src/utils/DTO/PostsResponse.js
class PostsResponse {
  constructor(post) {
    this.postId = post.postId;
    this.postTime = post.postTime;
    this.updateAt = post.updateAt;
    this.content = post.content || null;
    this.photos = post.photos || null;
    this.files = post.files || null;
    this.location = post.location || null;
    this.userId = post.userId || null;
    this.userName = post.userName || null;
    this.postGroupId = post.postGroupId || null;
    this.postGroupName = post.postGroupName || null;
    this.comments = post.comments || [];
    this.likes = post.likes || [];
    this.roleName = post.roleName || null;
    this.privacyLevel = post.privacyLevel || null;
    this.avatarUser = post.avatarUser || null;
  }
}

module.exports = PostsResponse;

File: group-service/src/main/java/vn/iostar/groupservice/dto/response/PostsResponse.java
package vn.iostar.groupservice.dto.response;

import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
public class PostsResponse {
	private String postId;
	private Date postTime;
	private Date updateAt;
	private String content;
	private String photos;
	private String files;
	private String location;
	private String userId;
	private String userName;
	private String avatarUser;
	private String postGroupId;
	private String postGroupName;
	private String avatarGroup;
	private String groupType;
	private List<String> comments;
	private List<String> likes;
}
